src/Datagrid/ProxyQuery.php

```diff
@@ -201,6 +202,21 @@ public function execute(array $params = [], $hydrationMode = null)
         }
 
         $query = $this->getDoctrineQuery();
+        $queryBuilder = $this->getQueryBuilder();
+
+        $identifierFieldNames = $queryBuilder
+            ->getEntityManager()
+            ->getMetadataFactory()
+            ->getMetadataFor(current($this->getQueryBuilder()->getRootEntities()))
+            ->getIdentifierFieldNames();
+        $hasSingleIdentifierName = 1 === \count($identifierFieldNames);
+        $hasJoins = 0 !== \count($queryBuilder->getDQLPart('join'));
```

Comment:
Maybe it's better to write
```suggestion
        $hasJoins = \count($queryBuilder->getDQLPart('join')) > 0;
```